src/main/java/org/ebyhr/trino/storage/StorageMetadata.java

```diff
@@ -144,38 +133,61 @@ public Map<SchemaTableName, List<ColumnMetadata>> listTableColumns(ConnectorSess
     public Iterator<TableColumnsMetadata> streamTableColumns(ConnectorSession session, SchemaTablePrefix prefix)
     {
         requireNonNull(prefix, "prefix is null");
-        return listTables(session, prefix).stream()
+        return listTables(session, prefix)
                 .map(table -> TableColumnsMetadata.forTable(
-                        table,
+                        table.toSchemaTableName(),
                         requireNonNull(getStorageTableMetadata(session, table), "tableMetadata is null")
                                 .getColumns()))
                 .iterator();
     }
 
-    private ConnectorTableMetadata getStorageTableMetadata(ConnectorSession session, SchemaTableName tableName)
+    private ConnectorTableMetadata getStorageTableMetadata(ConnectorSession session, CaseSensitiveSchemaTableName tableName)
     {
-        if (tableName.getSchemaName().equals(LIST_SCHEMA_NAME)) {
-            return new ConnectorTableMetadata(tableName, COLUMNS_METADATA);
+        if (tableName.schemaName().equals(LIST_SCHEMA_NAME)) {
+            return new ConnectorTableMetadata(tableName.toSchemaTableName(), COLUMNS_METADATA);
         }
 
-        if (!listSchemaNames().contains(tableName.getSchemaName())) {
+        if (!listSchemaNames().contains(tableName.schemaName())) {
             return null;
         }
 
-        StorageTable table = storageClient.getTable(session, tableName.getSchemaName(), tableName.getTableName());
+        StorageTable table = storageClient.getTable(session, tableName.schemaName(), tableName.tableName());
         if (table == null) {
             return null;
         }
 
-        return new ConnectorTableMetadata(tableName, table.getColumnsMetadata());
+        return new ConnectorTableMetadata(tableName.toSchemaTableName(), table.getColumnsMetadata());
     }
 
-    private List<SchemaTableName> listTables(ConnectorSession session, SchemaTablePrefix prefix)
+    /**
+     * Simplified variant of {@link SchemaTableName} that doesn't case-fold.
+     */
+    private static record CaseSensitiveSchemaTableName(String schemaName, String tableName)
+    {
+        /**
+         * Convert to {@link SchemaTableName}
+         *
+         * NOTE: This will case-fold the schema- and table names, so they may no longer be possible to resolve afterwards.
+         */
```

Comment:
Remove unhelpful javadoc. SchemaTableName is known as returning lowercase identifiers.

Reply:
I didn't know ... 

If the comment may be helpful to some people I'd say it does have value and could stay. 

It does explain an implementation detail that otherwise would need to be lookup up somewhere else, so is not a "increase var by 1" type of comment..

Reply:
I'd call it surprising misbehaviour but sure, it's your project, your call.